### DISCUSSION TITLE
Clash Tunel 是如何挂载上去的

### DISCUSSION THREAD

**Discussion by FoundTheWOUT** [zzzgydi/clash-verge#103] [Q&A] [answered]
看了一圈代码，只是看到了在 clash 内部有个 _tun_mod 方法，貌似就是在内部更新了一下配置（而且貌似就软件自己hold住，并没有更新到 config.yml ?）并没有调用 wintun.dll ，那 Clash tunnel 这个网络是怎么挂上去的。
而且服务模式和tun模式要同时开才能挂载。

**Comment by zzzgydi** [comment, marked as answer]
配置是在运行时合成的，从config.yaml和其他一些profile合成的。
调用wintun.dll是clash内核调用的，配置里开启tun就会有调用。
开tun模式需要管理员模式，所以开服务模式可以拉起UAC，或者在启动软件的时候用管理员模式启动。
